src/autoload/argparse.gd

```diff
@@ -0,0 +1,135 @@
+extends Node
+## Arguments are passed in in the `--double-dashed` format. After parse_args is called, these will
+## be converted to `snake_case` for easy processing.
+
+var used_arg_names: Array[String] = [] ## All the names of arguments that are taken.
+var switch_args: Array[SwitchArg] = [] ## Listing of all the possible switch arguments.
+var string_args: Array[StringArg] = [] ## Listing of all the possible string arguments.
+var int_args: Array[IntArg] = [] ## Listing of all the possible integer arguments.
+
+## The parsed command line arguments, in dictionary form. Populated by parse_args.
+var args: Dictionary = {}
+
+
+## Add a switch (boolean) argument.
+func add_switch_arg(arg_name: String) -> void:
+	if arg_name in used_arg_names:
+		push_error("Argument with identical name already exists.")
+		return
+	var arg := SwitchArg.new()
+	arg.name = arg_name
+	switch_args.append(arg)
+	used_arg_names.append(arg_name)
+
+
+## Add a string argument.
+func add_string_arg(arg_name: String, default: String = "", required: bool = false) -> void:
+	if arg_name in used_arg_names:
+		push_error("Argument with identical name already exists.")
+		return
+	var arg := StringArg.new()
+	arg.name = arg_name
+	arg.default = default
+	arg.required = required
+	string_args.append(arg)
+	used_arg_names.append(arg_name)
+
+
+## Add an integer argument.
+func add_int_arg(arg_name: String, default: int = 0, required: bool = false) -> void:
+	if arg_name in used_arg_names:
+		push_error("Argument with identical name already exists.")
+		return
+	var arg := IntArg.new()
+	arg.name = arg_name
+	arg.default = default
+	arg.required = required
+	int_args.append(arg)
+	used_arg_names.append(arg_name)
+
+
+## Parse the arguments from the command line. This should be once after all desired arguments are
+## added. Will return `true` if there were no errors, and `false` otherwise. The parsed args are
+## stored in the `args` class member.
+##
+## Note: arguments are passed in in the `--double-dashed` format. After parse_args is called, these
+## will be converted to `snake_case` for easy processing.
```

Comment:
Duplicated comment from the top of the file. Fine leaving, but wanted to call it out